torch/_dynamo/variables/lists.py

```diff
@@ -641,3 +646,45 @@ def reconstruct(self, codegen):
 
 class TupleIteratorVariable(ListIteratorVariable):
     pass
+
+
+class SetVariable(CommonListMethodsVariable):
+    def python_type(self):
+        return list
+
+    def reconstruct(self, codegen):
+        import sys
+
+        if sys.version_info >= (3, 11):
+            codegen.foreach(self.items)
+            return [create_instruction("BUILD_SET", arg=len(self.items))]
+        else:
+            create_load_global("set", True)
+            codegen.foreach(self.items)
+            return [create_instruction("CALL_FUNCTION", arg=len(self.items))]
+
+    def call_method(
+        self,
+        tx,
+        name,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        options = VariableTracker.propagate(self, args, kwargs.values())
+        if name == "add" and args:
+            assert not kwargs
+            item = args[0]
+            items = set(self.items)
+            items.add(item)
```

Comment:
item is a VT, which could represent a py constant or other type.

Is it gauranteed that we'll get the same list of items by using set(VT's).add(otherVT) as we would with set(pyobjs).add(other) for all types?

like i guess one simple case would be if we wrapped 2 different int constants with the value 2 in VT's at various points and we tried to jam them into this set- would they both be allowed to show up due to being different VT instances?

Reply:
Yea I think this needs better unpacking